py-api-proposal.md

```diff
@@ -0,0 +1,390 @@
+# Genkit Python API Proposal
+
+Open questions:
+
+- what's the best way to handle async? Should genkit primitive be all sync/blocking and leave it up to the user to deal with concurrency? I see some GenAI libs offer dual API -- sync and async (ex. [openai](https://github.com/openai/openai-python?tab=readme-ov-file#async-usage), [langchain](https://python.langchain.com/docs/concepts/async/)).
+
+## Init
+
+```py
+from genkit import Genkit
+from genkit.vertexai import vertexAI, gemini
+
+ai = Genkit(
+    plugins=[vertexAI()],
+    model=gemini("gemini-1.5-flash")
+)
+```
+
+## Flows
+
+Simplest flow:
+
+```py
+@ai.flow()
+def sayHi():
+    return "hi"
+```
+
+the flow name defaults to the function name, but can be overridden:
+
+```py
+@ai.flow(name="sayBye")
+def sayHi():
+    return "hi"
+```
+
+flows can take input:
+
+```py
+@ai.flow()
+def sayHi(name: str):
+    return f"hi {name}"
+```
+
+and can accept pydantic objects:
+
+```py
+from pydantic import BaseModel, Field
+
+class TwoNumbers(BaseModel):
+    a: int = Field(description="a field")
+    b: int = Field(description="b field")
+
+@ai.flow()
```

Comment:
Why not specify the `input_model` in the annotation? (Seems like output_model is being specified here)

Reply:
in python we rely on reflection. The type annotation on the function itself is our input/output model annotation.